src/plugins/discover/public/application/components/discover_legacy.tsx

```diff
@@ -152,6 +153,18 @@ export function DiscoverLegacy({
     'col-md-12': isSidebarClosed,
   });
 
+  const onTrackQuery = () => {
+    if (trackUiMetric) {
```

Comment:
I don't think you should make this a discover feature using callbacks.
Lets add those metrics directly into the `SearchBar` component and report them along side the current application id.